### PR TITLE
Add tests for batchdelete

### LINKED ISSUE #220: Add test for batchdelete command
Write JUnit tests to test for the `batchdelete` command

### PR DESCRIPTION
**Pull request by KumChaiYin**
Closes #220 

Add tests
Modify some previous code to solve bugs

### CODE REVIEW

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 1]
```diff
@@ -47,20 +47,20 @@ public class TypicalPersons {
...
-            .withPolicy("NTUC", "NOPOLICY", "01-01-1000", "01-01-1000").build();
+            .withPolicy("NOCOMPANY", "NOPOLICY", "01-01-1000", "01-01-1000").build();
     public static final Person FIONA = new PersonBuilder().withName("Fiona Kunz").withPhone("94824277")
             .withEmail("lydia@example.com").withAddress("little tokyo").withNric("743G").withLicencePlate("SBA1Z")
-            .withPolicy("InsureMe", "NOPOLICY", "01-01-1000", "01-01-1000").build();
+            .withPolicy("NOCOMPANY", "NOPOLICY", "01-01-1000", "01-01-1000").build();
     public static final Person GEORGE = new PersonBuilder().withName("George Best").withPhone("94824422")
             .withEmail("anna@example.com").withAddress("4th street").withNric("674G").withLicencePlate("SNM9735R")
-            .withPolicy("InsureMe", "NOPOLICY", "01-01-1000", "01-01-1000").build();
+            .withPolicy("NOCOMPANY", "NOPOLICY", "01-01-1000", "01-01-1000").build();
```
Comment (KumChaiYin) [inline src/test/java/seedu/address/testutil/TypicalPersons.java]: One thing here to note that is:
Although these all inputs seems like default policy,
when apply PersonBuilder, these clients still have policy (not default)
(but this does not affect other tests)

**Review by KumChaiYin — COMMENTED** [review]
Add a comment to noted

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -88,6 +89,14 @@ public class CommandTestUtil {
     public static final String INVALID_EMAIL_DESC = " " + PREFIX_EMAIL + "bob!yahoo"; // missing '@' symbol
     public static final String INVALID_ADDRESS_DESC = " " + PREFIX_ADDRESS; // empty string not allowed for addresses
     public static final String INVALID_TAG_DESC = " " + PREFIX_TAG + "hubby*"; // '*' not allowed in tags
+    public static final String INVALID_COMPANY_DESC = " " + PREFIX_TAG + ""; // company cannot be empty
```
Comment (cyaoxuan) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: Should be PREFIX_COMPANY?

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 3]
```diff
@@ -47,20 +47,20 @@ public class TypicalPersons {
...
-            .withPolicy("InsureMe", "5734534X", "31-12-2019", "01-01-2020").build();
+            .withPolicy("InsureMe", "5734534X", "31-12-2019", "01-02-2020").build();
     public static final Person DANIEL = new PersonBuilder().withName("Daniel Meier").withPhone("87652533")
             .withEmail("cornelia@example.com").withAddress("10th street").withTags("friends")
             .withNric("464J").withLicencePlate("STG46P")
             .withPolicy("NTUC", "4532", "29-02-2004", "29-02-2008").build();
     public static final Person ELLE = new PersonBuilder().withName("Elle Meyer").withPhone("94822244")
             .withEmail("werner@example.com").withAddress("michegan ave").withNric("573H").withLicencePlate("SHL463C")
-            .withPolicy("NTUC", "NOPOLICY", "01-01-1000", "01-01-1000").build();
+            .withPolicy("NOCOMPANY", "NOPOLICY", "01-01-1000", "01-01-1000").build();
```
Comment (cyaoxuan) [inline src/test/java/seedu/address/testutil/TypicalPersons.java]: Default value is now `!@#NO_COMPANY!@#` to make it more foolproof. Can you update and see if the tests get affected?

File: src/test/java/seedu/address/logic/commands/BatchDeleteCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        String expected2 = BatchDeleteCommand.class.getCanonicalName()
+                + "{toBatchDeleteMonth=[not applicable], toBatchDeleteCompany=" + company + "}";
+
+        assertEquals(expected2, command2.toString());
+    }
+
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    private class ModelStub implements Model {
```
Comment (cyaoxuan) [inline src/test/java/seedu/address/logic/commands/BatchDeleteCommandTest.java:144]: Nice use of stubs!

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 2 above)
Comment (KumChaiYin) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: I removed this constant as it wasn't being used

File: src/test/java/seedu/address/testutil/TypicalPersons.java (hunk 3 above)
Comment (KumChaiYin) [reply src/test/java/seedu/address/testutil/TypicalPersons.java]: Thanks for pointing it out!

**Pull request merged**